libc/src/time/ctime_s.cpp

```diff
@@ -0,0 +1,49 @@
+//===-- Implementation of ctime_s function --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "ctime_s.h"
+#include "hdr/errno_macros.h"
+#include "src/__support/CPP/limits.h"
+#include "src/__support/common.h"
+#include "src/__support/macros/config.h"
+#include "time_utils.h"
+
+namespace LIBC_NAMESPACE_DECL {
+
+using LIBC_NAMESPACE::time_utils::TimeConstants;
+
+LLVM_LIBC_FUNCTION(int, ctime_s,
+                   (char *buffer, rsize_t buffer_size, const time_t *t_ptr)) {
+  // TODO (https://github.com/llvm/llvm-project/issues/115907): invoke
+  // constraint handler
+  if (buffer == nullptr || t_ptr == nullptr)
+    return EINVAL;
```

Comment:
K.3.8.3.3.2:
> maxsize shall not be less than 26 and shall not be greater than RSIZE_MAX.

So we're missing a comparison against 26 (and a test for `buffer_size` being 26 or less).

Reply:
https://github.com/llvm/llvm-project/pull/110676/files#r1843606831